daemon/container.c

```diff
@@ -2157,8 +2160,21 @@ int
 container_add_net_iface(container_t *container, const char *iface, bool persistent)
 {
 	ASSERT(container);
+
+	IF_FALSE_RETVAL(cmld_netif_phys_remove_by_name(iface), -1);
+
 	pid_t pid = container_get_pid(container);
-	int res = c_net_move_ifi(iface, pid);
+	pid_t pid_a0 = container_get_pid(cmld_containers_get_a0());
+
+	/* if c0 is running the interfaces is occupied by c0, thus we have
+	 * to take it bake to cml first.
+	 */
+	container_state_t state_a0 = container_get_state(cmld_containers_get_a0());
+	int res = 0;
+	if (state_a0 == CONTAINER_STATE_RUNNING || state_a0 == CONTAINER_STATE_BOOTING)
+		res = c_net_remove_ifi(iface, pid_a0);
+
```

Comment:
better directly abort container start if if iface could not be removed to avoid race/information leak from container to c0/privileged container?

Reply:
this is not during container start. This function is called by control interface to add
netifs through runtime. For those cases, where this is an issue, we need to introduce a restricted control interface anyway. Maybe configurable through device.conf